src/formatDate.js

```diff
@@ -50,7 +50,38 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const oldYearIndex = fromFormat.indexOf('YY') === -1
+    ? fromFormat.indexOf('YYYY') : fromFormat.indexOf('YY');
+
+  const dateObj = {
+    day: dateArr[fromFormat.indexOf('DD')],
+    month: dateArr[fromFormat.indexOf('MM')],
+    year: dateArr[oldYearIndex],
+  };
+
+  const newDateArr = [];
+
+  newDateArr[toFormat.indexOf('DD')] = dateObj.day;
+  newDateArr[toFormat.indexOf('MM')] = dateObj.month;
+
+  const newYearIndex = toFormat.indexOf('YY') === -1
+    ? toFormat.indexOf('YYYY') : toFormat.indexOf('YY');
+
+  const currentYearFormat = dateObj.year.length;
+  const expectedYearFormat = toFormat[newYearIndex].length;
+
+  if (currentYearFormat > expectedYearFormat) {
```

Comment:
Good!
You avoid nesting `else if `conditions. Fine)

And also you can make separate variables for `if` conditions to make it again more readable